pkg/kube/client.go

```diff
@@ -346,6 +361,12 @@ func newClientInternal(clientFactory util.Factory, revision string) (*client, er
 	}
 	c.gatewayapiInformer = gatewayapiinformer.NewSharedInformerFactory(c.gatewayapi, resyncInterval)
 
+	c.mcsapis, err = mcsapisClient.NewForConfig(c.config)
+	if err != nil {
+		return nil, err
+	}
+	c.mcsapisInformers = mcsapisInformer.NewSharedInformerFactory(c.mcsapis, resyncInterval)
```

Comment:
This is done regardless of the environment variable. Will there be any impact on Istio when disabled?

Reply:
That's a good point - as per https://github.com/kubernetes/kubernetes/issues/79610 it at least will flood kube-controller logs which isn't great. 

I don't think my solution is the cleanest, so I'd welcome alternative suggestions. I just had no idea how to get the env variable value into the client creation other than checking it in the method itself

Reply:
Awesome. I'll undo that change